src/App.tsx

```diff
@@ -1,61 +1,154 @@
+import { FormEvent, useState } from 'react';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUserById(todoUserId: number): User | null {
+  const userObj = usersFromServer.find(user => user.id === todoUserId);
+
+  return userObj || null;
+}
+
+export const preparedTodos: Todo[] = todosFromServer.map((todo) => {
+  return {
+    id: todo.id,
+    title: todo.title,
+    userId: todo.userId,
+    completed: todo.completed,
+    user: getUserById(todo.userId),
+  };
+});
+
+function newTodoId(todos: Todo[]) {
+  let { id } = todos[0];
+
+  todos.forEach(todo => {
+    if (id < todo.id) {
+      id = todo.id;
+    }
+  });
+
+  return id;
+}
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userName, setOption] = useState('0');
```

Comment:
it's userId, not a userName
![image](https://github.com/mate-academy/react_add-todo-form/assets/91411692/59cb47a5-0d12-4fa8-beaf-a851cb256738)
```suggestion
  const [userId, setUserId] = useState('0');
```